addons/account/models/res_partner_bank.py

```diff
@@ -280,14 +280,17 @@ def write(self, vals):
         # leaves them vulnerable to edits via the shell/... So we need to ensure that the user has the rights to edit
         # these fields when writing too.
         # While we do lock changes if the account is trusted, we still want to allow to change them if we go from not trusted -> trusted or from trusted -> not trusted.
-        any_trusted_accounts = any(account.lock_trust_fields for account in self)
+        trusted_accounts = self.filtered(lambda x: x.lock_trust_fields)
+        any_trusted_accounts = bool(trusted_accounts)
         if not any_trusted_accounts:
             should_allow_changes = True  # If we were on a non-trusted account, we will allow to change (setting/... one last time before trusting)
         else:
             # If we were on a trusted account, we only allow changes if the account is moving to untrusted.
             should_allow_changes = ('allow_out_payment' in vals and vals['allow_out_payment'] is False)
 
-        if ('acc_number' in vals or 'partner_id' in vals) and not should_allow_changes:
+        lock_fields = {'acc_number', 'sanitized_acc_number', 'acc_holder_name', 'partner_id', 'company_id', 'bank_id', 'acc_type', 'active', 'currency_id'}
+        updated_lock_fields = lock_fields & set(vals.keys())
+        if not should_allow_changes and updated_lock_fields and any(any(account[u] != vals[u] for u in updated_lock_fields) for account in trusted_accounts):
             raise UserError(_("You cannot modify the account number or partner of an account that has been trusted."))
```

Comment:
This line needs to be updated accordingly.

Reply:
I don't understand your comment.

Reply:
The line mentions `account number` and `partner` because previously, only 'acc_number' and 'partner_id' were being checked. Now it checks more things, then the error message is outdated. I think the error message should mention exactly what field is being changed that triggers the error.